src/Messages/Interactions/TSErrorMessage.m

```diff
@@ -36,8 +32,14 @@ - (instancetype)initWithTimestamp:(uint64_t)timestamp
     }
 
     _errorType = errorMessageType;
-    // TODO Move this out of model class.
-    [[TextSecureKitEnv sharedEnv].notificationsManager notifyUserForErrorMessage:self inThread:thread];
+    // TODO: Move this out of model class.
+    //
+    //       For now, dispatch async to ensure we're not inside a transaction
+    //       and thereby avoid deadlock.
+    TSErrorMessage *errorMessage = self;
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
```

Comment:
`notifyUserForErrorMessage: inThread :` eventually uses a transaction within `PropertyListPreferences`, which caused deadlock if you created any `TSErrorMessage` within a transaction.